src/TestData/Routes/DonationsRoute.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace GiveTestData\TestData\Routes;
+
+use WP_REST_Request;
+use WP_REST_Response;
+use Throwable;
+use GiveTestData\TestData\Factories\DonationFactory as DonationFactory;
+use GiveTestData\TestData\Repositories\DonationRepository as DonationRepository;
+
+/**
+ * Class FundOverviewRoute
+ * @package GiveFunds\Routes
+ */
+class DonationsRoute extends Endpoint {
+
+	/**
+	 * Maximum number of donations to generate per request
+	 * @var int
+	 */
+	private $limit = 30;
+
+	/** @var string */
+	protected $endpoint = 'give-test-data/generate-donations';
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerRoute() {
+		register_rest_route(
+			'give-api/v2',
+			$this->endpoint,
+			[
+				[
+					'methods'             => 'POST',
+					'callback'            => [ $this, 'handleRequest' ],
+					'permission_callback' => [ $this, 'permissionsCheck' ],
+					'args'                => [
+						'status'  => [
+							'type'              => 'string',
+							'required'          => true,
+							'validate_callback' => [ $this, 'validateDonationStatus' ],
+						],
+						'count'   => [
+							'type'              => 'integer',
+							'required'          => true,
+							'sanitize_callback' => [ $this, 'sanitizeNumber' ],
+						],
+						'revenue' => [
+							'type'              => 'integer',
+							'required'          => false,
+							'sanitize_callback' => [ $this, 'sanitizeNumber' ],
+						],
+					],
+				],
+				'schema' => [ $this, 'getSchema' ]
+			]
+		);
+	}
+
+	/**
+	 * @return array
+	 * @since 1.0.0
+	 */
+	public function getSchema() {
+		return [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'give-funds',
+			'type'       => 'object',
+			'properties' => [
+				'status'  => [
+					'type'        => 'string',
+					'description' => esc_html__( 'Donation status', 'give-test-data' ),
+				],
+				'count'   => [
+					'type'        => 'integer',
+					'description' => esc_html__( 'Number of donations to generate', 'give-test-data' ),
+				],
+				'revenue' => [
+					'type'        => 'integer',
+					'description' => esc_html__( 'Revenue amount', 'give-test-data' ),
+				],
+			],
+		];
+	}
+
+	/**
+	 * @param WP_REST_Request $request
+	 *
+	 * @return WP_REST_Response
+	 * @since 1.0.0
+	 */
+	public function handleRequest( WP_REST_Request $request ) {
+		global $wpdb;
+
+		$donationFactory    = give( DonationFactory::class );
+		$donationRepository = give( DonationRepository::class );
```

Comment:
These dependencies should be injected via the Constructor. This will help with any automated testing, which can easily inject a mock.